cellprofiler_core/utilities/hdf5_dict.py

```diff
@@ -386,8 +384,23 @@ def flush(self):
     def file_contents(self):
         with self.lock:
             self.flush()
-            with open(self.filename, "rb") as f:
-                return memoryview(f.read())
+            try:
+                with open(self.filename, "rb") as f:
+                    return memoryview(f.read())
+            except PermissionError:
+                self.hdf5_file.close()
```

Comment:
Should probably to `self.close()` here instead

Reply:
Yeah, I think I wasn't initially remaking the top_group, which is why I went this way, but now that we are, good call.

Reply:
Nope, can't do that, it ends up deleting the file if it's a temp file, so now all the tests are failing, and any condition where the user would be using a temp file would have the same issue.